src/NumberGuessingGame.js

```diff
@@ -1,77 +1,53 @@
-import React, { Component } from "react";
+import React, { useState } from "react";
 import GuessControl from "./GuessControl";
 import GuessMessage from "./GuessMessage";
 import GameOver from "./GameOver";
 
-/**
- *
- * Returns a random integer number from 1-100 inclusive
- */
-function getRandomNumber() {
-  return Math.floor(Math.random() * 100) + 1;
-}
-
-const MAX_ATTEMPTS = 5;
-
-class NumberGuessingGame extends Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    };
-
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleGuess = this.handleGuess.bind(this);
-    this.handleReset = this.handleReset.bind(this);
-  }
-
-  handleGuess(guess) {
-    this.setState({
-      latestGuess: guess,
-      numberOfGuesses: this.state.numberOfGuesses + 1,
-    });
+const NumberGuessingGame = () => {
+  /**
+   *
+   * Returns a random integer number from 1-100 inclusive
+   */
+  function getRandomNumber() {
+    return Math.floor(Math.random() * 100) + 1;
   }
 
-  handleReset() {
-    this.setState({
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    });
-  }
-
-  render() {
-    const isCorrectGuess = this.state.latestGuess === this.state.numberToGuess;
-
-    const isGameOver =
-      isCorrectGuess || this.state.numberOfGuesses === MAX_ATTEMPTS;
-
-    return (
-      <div>
-        <h2>I'm thinking of a number from 1 to 100.</h2>
-        <h2>
-          Can you guess the number I am thinking of in {MAX_ATTEMPTS} tries?
-        </h2>
-        <GuessControl onGuess={this.handleGuess} />
-        {isGameOver && (
-          <GameOver hasWon={isCorrectGuess} onReset={this.handleReset} />
-        )}
-        {!isGameOver && (
-          <GuessMessage
-            guess={this.state.latestGuess}
-            numberToGuess={this.state.numberToGuess}
-            numberOfGuesses={this.state.numberOfGuesses}
-          />
-        )}
-      </div>
-    );
-  }
-}
+  const MAX_ATTEMPTS = 5;
+  const [numberToGuess, setNumberToGuess] = useState(getRandomNumber());
+  const [numberOfGuesses, setNumberOfGuesses] = useState(0);
+  const [latestGuess, setLatestGuess] = useState(null);
```

Comment:
Your approach works, but let me give you another way to do it...
When you have similar and many related states you could create a unique state with an object shape, something like:

```
const [guessData, setGuessData] = useState({
  numberToGuess: getRandomNumber(),
  numberOfGuesses: 0,
  latestGuess: null,
});
```
With that you can factor out that state initial value in a variable and do this:
```
const initialGuessData = {
  numberToGuess: getRandomNumber(),
  numberOfGuesses: 0,
  latestGuess: null,
};
const [guessData, setGuessData] = useState(initialGuessData);
```
and reuse it when handling reset:
 ```
 const handleReset = () => {
    setData(initialData);
  };
```
The only "tricky" part would be setting the guessData (using object destructuring):
```
  const handleGuess = (guess) => {
    setGuessData({
      ...guessData,
      latestGuess: guess,
      numberOfGuesses: guessData.numberOfGuesses + 1,
    });
  };
```
You'll get 3 mayor benefits when doing that:
1. Less and cleaner code.
2. You could refactor code (initialGuessData)
3. In the web developer tools if you've got the "React Developer Tools extensions" for Chrome, you'll have a better time debugging your App state.

Reply:
I like it!